### PR TITLE
[Gradle] - Update dependencies

### PR DESCRIPTION
**Pull request by jaredsburrows**
@benjamin-bader 
 - Upgraded dependencies
 - Updated to use new Java plugin from Gradle
 - Removed left over semi-colons from `.groovy` files
 - For some reason, when upgrading the Android Gradle plugin, I cannot seem to use "apply" for the tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,12 @@
 language: android
 
-groovy:
```
Comment (jaredsburrows) [inline .travis.yml]: Not needed.

File: integration/tests/build.gradle [hunk 2]
```diff
@@ -27,6 +27,5 @@ android {
 dependencies {
     // Android Testing Support Library"s runner and rules and hamcrest matchers
     compile deps.runner
-    compile deps.rules
```
Comment (jaredsburrows) [inline integration/tests/build.gradle]: This "integration/tests" project is empty, should we just remove it?

File: .travis.yml (hunk 1 above)
Comment (benjamin-bader) [inline .travis.yml]: Well, I'll be damned.  I no longer remember why I thought this would create a build matrix in Travis, but I can't find any trace in their github that it ever worked.

Too bad - 2.3.9 introduced a breaking change of some sort that we worked around.  It's less and less relevant as time goes by, but it would be nice to have test coverage there.

File: integration/tests/build.gradle (hunk 2 above)
Comment (benjamin-bader) [inline integration/tests/build.gradle]: Yes, I think that would be best.

File: .travis.yml (hunk 1 above)
Comment (jaredsburrows) [inline .travis.yml]: We should just be using the current Gradle version of Groovy: `localGroovy()`.

File: integration/tests/build.gradle (hunk 2 above)
Comment (jaredsburrows) [inline integration/tests/build.gradle]: Removed.

File: src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountTask.groovy [hunk 3]
```diff
@@ -350,7 +348,6 @@ class ModernMethodCountTask extends DexMethodCountTaskBase {
     }
 }
 
-@ParallelizableTask
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountTask.groovy]: This was removed in Gradle 4.0 - https://discuss.gradle.org/t/parallelizabletask-is-removed-in-gradle-4-0/22970/10.

File: build.gradle [hunk 4]
```diff
@@ -54,14 +54,14 @@ jar {
 }
 
 dependencies {
-    compile gradleApi()
-    compile deps.gson
+    implementation localGroovy()
```
Comment (jaredsburrows) [inline build.gradle]: Newer Java gradle plugin - https://docs.gradle.org/current/userguide/java_library_plugin.html.

**Comment by jaredsburrows** [comment]
@benjamin-bader It looks like TravisCI has an issue? It cannot find the `android` binary. Made an issue here: https://github.com/travis-ci/travis-ci/issues/8123.

**Comment by jaredsburrows** [comment]
@benjamin-bader Looks like a breaking change from TravisCI? https://github.com/travis-ci/travis-ci/issues/5990

File: .travis.yml (hunk 1 above)
Comment (benjamin-bader) [inline .travis.yml]: In practice, we are - that's how the bug surfaced in the first place!

The problem is that different versions expose slightly different APIs.  In this case, `Collections.forEach` didn't exist prior to 2.3.9.  Testing against an older version would (in theory) prevent regressions, though it seems like this never worked to begin with.

File: build.gradle (hunk 4 above)
Comment (benjamin-bader) [inline build.gradle]: I've used the java-library plugin in thrifty, but it never occurred to me to use it here.  Thanks for updating!

Also, why did you remove `gradleApi()` - is that included somehow via `localGroovy()`?

Comment (jaredsburrows) [inline build.gradle]: @benjamin-bader The newer java library plugin automatically adds it.

File: .travis.yml (hunk 1 above)
Comment (jaredsburrows) [inline .travis.yml]: @benjamin-bader The "groovy" block in the `.travis.yml` file is for installing `groovy` on the build and using it locally(eg. groovy file.groovy). We are using Groovy as a language through Gradle. I do not think you need to install `groovy` locally in order to compile Groovy with Gradle unlike Java. You need to have Java installed to run and compile with Gradle. I could be wrong about Groovy.

File: src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountPlugin.groovy [hunk 5]
```diff
@@ -71,7 +71,7 @@ class DexMethodCountPlugin implements Plugin<Project> {
             return false
         }
 
-        // Java version strings are something like 1.8.0_65; we don't
+        // Java version strings are something like 1.8.0_65, we don't
```
Comment (benjamin-bader) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountPlugin.groovy]: nit: the semicolons in comments were correct and intentional; I'd prefer not to change them :)

Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountPlugin.groovy]: Oh sorry! Let me put them back.

Comment (benjamin-bader) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountPlugin.groovy]: TIL.  No big loss, as generally by the time this task is run all other work is finished.

File: build.gradle (hunk 4 above)
Comment (benjamin-bader) [inline build.gradle]: Oh, I missed that the plugin you're talking about is `java-gradle-plugin`, I thought it was `java-library-plugin`.  Makes sense!

File: .travis.yml (hunk 1 above)
Comment (benjamin-bader) [inline .travis.yml]: No, you're right.  I guess I thought it would contribute to the travis build matrix; `scala` in Travis does work that way, but `groovy` doesn't.

Anyways, good catch - thanks!

File: gradle/versions.gradle [hunk 6]
```diff
@@ -23,16 +22,16 @@ ext.plugs = [
...
-        // androidTestCompile
-        "runner"      : "com.android.support.test:runner:0.4.1",
-        "rules"       : "com.android.support.test:rules:0.4.1",
+        "gson"        : "com.google.code.gson:gson:2.8.1",
 
         // testCompile
         "junit"       : "junit:junit:4.12",
         "spockCore"   : "org.spockframework:spock-core:1.1-groovy-2.4",
+
+        // integration - if you change these, you need to update the values in ./.buildscript/test.sh
```
Comment (jaredsburrows) [inline gradle/versions.gradle]: @benjamin-bader Added this comment to help future developers to ensure these dependencies remain untouched. If they are touched, the developer needs to update the current dex count values in the `test.sh`.

File: integration/tests/build.gradle [hunk 7]
```diff
@@ -18,7 +18,6 @@ android {
     // Set the target app project. The module specified here should contain the production code
     // test should run against.
     targetProjectPath ":app"
-    targetVariant "debug"
```
Comment (jaredsburrows) [inline integration/tests/build.gradle]: This is now deprecated:
```
android.targetVariant is deprecated, all variants are now tested.
```

**Comment by jaredsburrows** [comment]
@benjamin-bader Fixed this PR as well.
